File: app/src/androidTest/java/za/co/duartbreedt/androidconceptsreference/MainActivityTest.java
package za.co.duartbreedt.androidconceptsreference;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
import static androidx.test.espresso.action.ViewActions.typeText;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;

import androidx.test.ext.junit.rules.ActivityScenarioRule;
import androidx.test.ext.junit.runners.AndroidJUnit4;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4.class)
public class MainActivityTest {

    @Rule
    public ActivityScenarioRule<MainActivity> activityScenarioRule = new ActivityScenarioRule<>(MainActivity.class);

    @Test
    public void inputNameAndNavigate() {

        // MainActivity.java views
        onView(withId(R.id.editText_username)).perform(typeText("Steve"), closeSoftKeyboard());
        onView(withId(R.id.button_navigate)).perform(click());

        // NextActivity.java views
        onView(withId(R.id.text_greeting)).check(matches(withText("Welcome!")));
    }
}

File: app/src/androidTest/java/za/co/duartbreedt/androidconceptsreference/MainActivityTest.kt
package za.co.duartbreedt.androidconceptsreference

import androidx.test.espresso.Espresso
import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.assertion.ViewAssertions
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.ext.junit.rules.ActivityScenarioRule
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MainActivityTest {

    @Rule
    var activityScenarioRule = ActivityScenarioRule(
        MainActivity::class.java
    )

    @Test
    fun inputNameAndNavigate() {

        // MainActivity.java views
        Espresso.onView(withId(R.id.editText_username))
            .perform(ViewActions.typeText("Steve"), ViewActions.closeSoftKeyboard())
        Espresso.onView(withId(R.id.button_navigate)).perform(ViewActions.click())

        // NextActivity.java views
        Espresso.onView(withId(R.id.text_greeting))
            .check(ViewAssertions.matches(ViewMatchers.withText("Welcome!")))
    }
}